scraper/src/mindtouch2zim/entrypoint.py

```diff
@@ -228,36 +208,21 @@ def main(tmpdir: str) -> None:
         help="Contact information to pass in User-Agent headers",
     )
 
-    args = parser.parse_args()
-
-    try:
-        init_context_from_args(args, tmpdir)
-
-        zim_config = ZimConfig.of(args)
-        content_filter = ContentFilter.of(args)
-
-        # remove trailing slash which we do not want per convention
-        CONTEXT.library_url = CONTEXT.library_url.rstrip("/")
+    args = parser.parse_args(raw_args)
 
-        # initialize all paths, ensuring they are ok for operation
-        CONTEXT.output_folder.mkdir(exist_ok=True)
-        validate_folder_writable(CONTEXT.output_folder)
+    # Ignore unset values so they do not override the default specified in Context
+    args_dict = {key: value for key, value in args._get_kwargs() if value}
 
-        CONTEXT.tmp_folder.mkdir(exist_ok=True)
-        validate_folder_writable(CONTEXT.tmp_folder)
+    # initialize some context properties that are "dynamic" (i.e. not constant
+    # values like an int, a string, ...)
+    if not args_dict.get("tmp_folder", None):
+        if tmp_from_env := os.getenv("MINDTOUCH_TMP"):
```

Comment:
I dont like this `os.getenv()` outside of the constants. Can we use a constant instead? Or a Context constant?

Reply:
Moved to Context constant